template.go

```diff
@@ -176,16 +177,27 @@ const (
 )
 
 // setFieldExpr returns a Go expression to set the field on a response.
-func setFieldExpr(f *gen.Field, ident string) (string, error) {
+func setFieldExpr(f *gen.Field, schema, ident string) (string, error) {
 	if !f.Optional {
-		return fmt.Sprintf("%s: %s.%s", f.StructField(), ident, f.StructField()), nil
+		expr := fmt.Sprintf("%s.%s", ident, f.StructField())
+		if f.IsEnum() {
+			expr = convertTo(schema+f.StructField(), expr)
```

Comment:
I think it won't work if you have a custom `GoType`, and need to use `gen.Field.Type.String`.

Reply:
It actually does work with `GoType` on enums. But `GoType`s don't work on other stuff. But that is to be fixed in another PR.

Reply:
Hey @chaychoong, can you create an issue and provide me an example of the type you are using?

Reply:
nvm, I found it. Will have a fix out soon.